src/lib/storage/segment_access_counter.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <array>
+#include <atomic>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "storage/pos_list.hpp"
+#include "types.hpp"
+
+namespace opossum {
+class Table;
+
+// The SegmentAccessCounter is a collection of counters to register how often and how a segment is accessed.
+class SegmentAccessCounter {
+  friend class SegmentAccessCounterTest;
+
+ public:
+  using CounterType = std::atomic_uint64_t;
+
+  enum class AccessType {
+    Point /* Single point access */,
+    Sequential /* 0, 1, 1, 2, 3, 4 */,
+    Increasing /* 0, 0, 1, 2, 4, 8, 17 */ /* or decreasing */,
+    Random /* 0, 1, 0, 42 */,
+    Dictionary,
```

Comment:
This could use a comment, too. People might be surprised to see a pos list above and dictionary here.